SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayInternalClient.java

```diff
@@ -0,0 +1,164 @@
+                            // SPAY_NOT_SUPPORTED
+                            error = SamsungPayError.SAMSUNG_PAY_NOT_SUPPORTED;
+                        }
+                    }
+                    samsungPayError = new SamsungPayException(error);
+                }
+
+                callback.onResult(statusCode, samsungPayError);
+            }
+
+            @Override
+            public void onFail(int errorCode, Bundle bundle) {
+                SamsungPayException exception = new SamsungPayException(errorCode);
+                callback.onResult(null, exception);
+            }
+        });
+    }
+
+    void getAcceptedCardBrands(final GetAcceptedCardBrandsCallback callback) {
+        paymentManager.requestCardInfo(new Bundle(), new PaymentManager.CardInfoListener() {
+            @Override
+            public void onResult(final List<CardInfo> cardInfos) {
+                Set<SpaySdk.Brand> spayAcceptedCardBrands = new HashSet<>();
+                if (cardInfos != null) {
+                    for (CardInfo cardInfo : cardInfos) {
+                        spayAcceptedCardBrands.add(cardInfo.getBrand());
+                    }
+                }
+                // equivalent to getting the intersection of both sets
+                spayAcceptedCardBrands.retainAll(braintreeAcceptedCardBrands);
+                callback.onResult(new ArrayList<>(spayAcceptedCardBrands), null);
+            }
+
+            @Override
+            public void onFailure(int errorCode, Bundle bundle) {
+                Exception error = new SamsungPayException(errorCode);
+                callback.onResult(null, error);
+            }
+        });
+    }
+
+    void startSamsungPay(CustomSheetPaymentInfo customSheetPaymentInfo, final SamsungPayStartListener listener) {
+        paymentManager.startInAppPayWithCustomSheet(customSheetPaymentInfo, new PaymentManager.CustomSheetTransactionInfoListener() {
+            @Override
+            public void onCardInfoUpdated(CardInfo cardInfo, CustomSheet customSheet) {
+                paymentManager.updateSheet(customSheet);
+                listener.onSamsungPayCardInfoUpdated(cardInfo, customSheet);
+            }
+
+            @Override
+            public void onSuccess(CustomSheetPaymentInfo customSheetPaymentInfo, String s, Bundle bundle) {
+                try {
+                    JSONObject json = new JSONObject(s);
+                    SamsungPayNonce samsungPayNonce = SamsungPayNonce.fromJSON(json);
+                    listener.onSamsungPayStartSuccess(samsungPayNonce, customSheetPaymentInfo);
+                } catch (JSONException e) {
+                    listener.onSamsungPayStartError(e);
+                }
+            }
+
+            @Override
+            public void onFailure(int errorCode, Bundle bundle) {
+                if (errorCode == SpaySdk.ERROR_USER_CANCELED) {
+                    UserCanceledException userCanceledError = new UserCanceledException("User Canceled");
```

Comment:
```suggestion
                    UserCanceledException userCanceledError = new UserCanceledException("User canceled Samsung Pay.");
```

nitpick just to match the other payment flows.